api/plugins/action/deploy_target_config.py

```diff
@@ -24,44 +17,67 @@ def run(self, tmp=None, task_vars=None):
 
         self.createClient(task_vars)
 
-        display.vvv("With: " + self._task.args.get('project'))
-        lagoonProject = Project(self.client).byName(self._task.args.get('project')).withDeployTargetConfigs()
+        self._display.vvv(f"With: {self._task.args}")
+        lagoonProject = Project(self.client).byName(
+            self._task.args.get('project'),
+            ['id', 'name']).withDeployTargetConfigs()
 
         if len(lagoonProject.errors) > 0:
             result['failed'] = True
             return result
 
         state = self._task.args.get('state', 'present')
         replace = self._task.args.get('replace', False)
+        configs = self._task.args.get('configs', [])
 
         for project in lagoonProject.projects:
             if state == "present":
                 changes = determine_required_updates(
                     project["deployTargetConfigs"],
-                    self._task.args.get('configs', []),
+                    configs,
                 )
                 result['changed'] = False
                 if len(changes) > 0:
                     for config in changes:
-                        if replace:
-                            DeployTargetConfig(self.client).delete(project['id'], changes['_existing_id'])
-
-                        if DeployTargetConfig(self.client).add(
-                                int(project['id']), 
-                                config['branches'],
-                                int(config['deployTarget']),
-                                config['pullrequests'],
-                                int(config['weight']) if 'weight' in config.keys() else 0
-                            ):
-                            result['result'].append(config)
+                        if replace and config.get('_existing_id'):
```

Comment:
When a new config is provided along with existing ones, this was failing because there's no `_existing_id` for that config.